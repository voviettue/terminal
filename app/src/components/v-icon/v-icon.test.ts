import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import VIcon from './v-icon.vue';

test('Mount component', () => {
	expect(VIcon).toBeTruthy();

	const wrapper = mount(VIcon, {
		props: {
			name: 'close',
		},
	});

	expect(wrapper.html()).toMatchSnapshot();
});

test('style props', () => {
	const props = ['x-small', 'small', 'large', 'x-large', 'left', 'right'];

	for (const prop of props) {
		const wrapper = mount(VIcon, {
			props: {
				name: 'close',
				[prop]: true,
			},
		});

		expect(wrapper.classes()).toContain(prop);
	}
});

test('custom icon', () => {
	const wrapper = mount(VIcon, {
		props: {
			name: 'signal_wifi_3_bar',
		},
	});

	expect(wrapper.find('svg').exists()).toBeTruthy();
});

test('social icon', () => {
	const wrapper = mount(VIcon, {
		props: {
			name: 'docker',
		},
	});

	expect(wrapper.find('svg').exists()).toBeTruthy();
});
