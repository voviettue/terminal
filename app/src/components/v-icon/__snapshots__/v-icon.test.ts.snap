// Vitest Snapshot v1

exports[`Mount component 1`] = `"<span class=\\"v-icon\\" data-v-4b41578f=\\"\\"><i class=\\"\\" data-icon=\\"close\\" data-v-4b41578f=\\"\\"></i></span>"`;
